galsim/base.py

```diff
@@ -2892,6 +2892,119 @@ def __str__(self):
+    R_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk.
+    
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+    
+    At present, this profile is not enabled for photon-shooting.
+    
+    Initialization
+    --------------
+
+    @param i                The inclination angle, i
+    @param scale_radius     The scale radius of the exponential disk.  Typically given in arcsec.
+    @param scale_height     The scale height of the exponential disk.  Typically given in arcsec.
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, InclinedExponential has the following access methods:
+
+        >>> i = inclined_exponential_obj.getI()
+        >>> r0 = inclined_exponential_obj.getScaleRadius()
+        >>> h0 = inclined_exponential_obj.getScaleHeight()
+    """
+    _req_params = { "i" : float, "scale_radius" : float, "scale_height" : float }
+    _opt_params = { "flux" : float }
+    _takes_rng = False
+
+    def __init__(self, i, scale_radius, scale_height,
+                 flux=1., gsparams=None):
+        
+        if not isinstance(i, galsim.Angle):
+            raise TypeError("Input i should be an Angle")
+        
+        i_rad = i.rad()
+        
+        GSObject.__init__(self, _galsim.SBInclinedExponential(i_rad, scale_radius, scale_height, flux,
+                                                              gsparams))
+        self._gsparams = gsparams
+
+    def getI(self):
+        """Return the inclination angle 'i' for this profile in radians.
+        """
+        return self.SBProfile.getI()
+
+    def getScaleRadius(self):
+        """Return the scale radius for this Sersic profile.
+        """
+        return self.SBProfile.getScaleRadius()
+
+    def getScaleHeight(self):
+        """Return the scale radius for this Sersic profile.
+        """
+        return self.SBProfile.getScaleHeight()
+
+    @property
+    def i(self): return self.getI()
+    @property
+    def scale_radius(self): return self.getScaleRadius()
+    @property
+    def scale_height(self): return self.getScaleHeight()
+
+    def __eq__(self, other):
+        return ((isinstance(other, galsim.InclinedExponential) and
+                 (self.i == other.i) and
+                 (self.scale_radius == other.scale_radius) and
+                 (self.scale_height == other.scale_height) and
+                 (self.flux == other.flux) and
+                 (self._gsparams == other._gsparams))
+                or
+                (isinstance(other, galsim.Exponential) and
```

Comment:
I don't think we want to declare an `InclinedExponential` with `i==0` "equal" to an `Exponential`.  Just keep the first part of this function.